src/tools/imgtool/imgtool.cpp

```diff
@@ -850,19 +839,19 @@ bool imgtool_validitychecks(void)
 	{
 		features = imgtool_get_module_features(module.get());
 
-		if (!module->name)
+		if (module->name.empty())
 		{
-			util::stream_format(std::wcerr, L"imgtool module %s has null 'name'\n", wstring_from_utf8(module->name));
+			util::stream_format(std::wcerr, L"imgtool module %s has null 'name'\n", wstring_from_utf8(module->name.c_str()));
```

Comment:
Why are you calling `.c_str()` for every call to `wstring_from_utf8`?  The prototype at src/lib/util.h:71 takes a `std::string` reference as a parameter.  (Yes, it should be changed to a string view, but that won’t stop a `std::string` being passed directly.)